sky/serve/serve_utils.py

```diff
@@ -1180,6 +1182,14 @@ def cluster_is_up() -> bool:
     log_prompt = re.match(_SKYPILOT_LOG_PATTERN, line)
 
     def _stream_provision_path(p: pathlib.Path) -> Iterator[str]:
+        # Check if this provision log has already been expanded
+        # to avoid duplicate expansion
```

Comment:
We should add the explanation for why there could be duplicated streaming here, i.e. talk about the rich spinner issue.